src/back/TicketManager/tests/Core/TicketManager.Core.DomainTests/Events/EventBuilderTests.cs

```diff
@@ -0,0 +1,287 @@
+using FluentAssertions;
+using TicketManager.Core.Domain.Events;
+using Xunit;
+
+namespace TicketManager.Core.DomainTests.Events;
+
+public class EventBuilderTests
+{
+    private readonly List<EventBuilder.SectorData> sectors;
+    private readonly Guid organizerId = Guid.NewGuid();
+    private readonly string name = "name";
+    private readonly string description = "description";
+    private readonly string location = "location";
+    private readonly DateTime date = DateTime.UtcNow.AddDays(1);
+
+    public EventBuilderTests()
+    {
+        sectors = new()
+        {
+            new("name 1", 5, 10, 12),
+            new("name 2", 10, 1, 1),
+            new("name 3", 123, 1123, 123),
+            new("name 4", 4, 13, 3),
+        };
+    }
+
+    [Fact]
+    public void WhenCorrectDataIsProvided_ItShouldReturnEvent()
+    {
+        var @event = new EventBuilder()
+            .WithGeneratedId()
+            .WithOrganizerId(organizerId)
+            .WithName(name)
+            .WithDescription(description)
+            .WithLocation(location)
+            .WithDate(date)
+            .WithSectors(sectors)
+            .Build();
+
+        @event.OrganizerId.Should().Be(organizerId);
+        @event.Name.Should().Be(name);
+        @event.Description.Should().Be(description);
+        @event.Location.Should().Be(location);
+        @event.Date.Should().Be(date);
+        @event.Sectors.Should().BeEquivalentTo(sectors);
+        @event.Sectors.Should().Match(s => s.All(s => s.EventId == @event.Id));
+    }
+    
+    [Fact]
+    public void WhenOrganizerIdIsNotProvided_ItShouldThrowArgumentException()
+    {
+        var func = () => new EventBuilder()
+            .WithGeneratedId()
+            .WithName(name)
+            .WithDescription(description)
+            .WithLocation(location)
+            .WithDate(date)
+            .WithSectors(sectors)
+            .Build();
+
+        func.Should().Throw<ArgumentException>();
```

Comment:
Shouldn't we specify which ArgumentException is thrown? In current form something can be messed up and for example `"Description is required"` could be thrown here if `Validate()` function was not set up properly